core/system/src/ipc.c

```diff
@@ -317,6 +335,13 @@ void ipc_drain_queue(void)
             continue;
         }
 
+        if (!ipc_io_token_is_ok(recv_ipc, recv_io)) {
+            /* The token of the IO request is invalid. Ignore it, don't put it
+             * back. This shouldn't happen in a non-malicious box. */
+            put_it_back(send_queue, send_slot);
```

Comment:
"Don't put it back" is for the receiving request. Will fix the comment at a later time (possibly those similar ones as well)